src/hotspot/cpu/x86/register_x86.hpp

```diff
@@ -152,16 +159,17 @@ class XMMRegisterImpl: public AbstractRegisterImpl {
   };
 
   // construction
-  friend XMMRegister as_XMMRegister(int encoding);
+  friend constexpr XMMRegister as_XMMRegister(int encoding);
 
-  inline VMReg as_VMReg();
+  inline VMReg as_VMReg() const;
 
   // derived registers, offsets, and addresses
   XMMRegister successor() const                          { return as_XMMRegister(encoding() + 1); }
 
   // accessors
-  int   encoding() const                          { assert(is_valid(), "invalid register (%d)", (int)(intptr_t)this ); return (intptr_t)this; }
-  bool  is_valid() const                          { return 0 <= (intptr_t)this && (intptr_t)this < number_of_registers; }
+  int raw_encoding() const                       { return this - first(); }
+  int   encoding() const                          { assert(is_valid(), "invalid register (%d)", (int)raw_encoding() ); return raw_encoding(); }
```

Comment:
In the other cases, we don't do this kind of printing assert. Stick to one style? I think it is fine to have a shorter assert, because `!is_valid()` basically implies `noreg`?

Reply:
OK.